app/src/layouts/appLayout/header/header.jsx

```diff
@@ -1,19 +1,109 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import classNames from 'classnames/bind';
-import { Link } from 'react-router-dom';
+import { FormattedMessage } from 'react-intl';
+import { Link, withRouter } from 'react-router-dom';
+import { userSelector } from 'controllers/auth';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
 import styles from './header.scss';
 import { ProjectSelector } from './projectSelector';
 
 const cx = classNames.bind(styles);
 
-export const Header = () => (
-  <div className={cx('header')}>
-    <Link to="/default_project/members">Members</Link>
-    <Link to="/default_project/settings">Settings</Link>
-    <Link to="/api">Api</Link>
-    <Link to="/administrate">Administrate</Link>
-    <Link to="/user-profile">Profile</Link>
-    <Link to="/login">Logout</Link>
-    <ProjectSelector projects={['Default_active', 'Default_active_2']} activeProject="Default_active" />
-  </div>
-);
+@withRouter
+@connect(state => ({
+  user: userSelector(state),
+}))
+export class Header extends PureComponent {
+  static propTypes = {
+    location: PropTypes.shape({
+      hash: PropTypes.string,
+      pathname: PropTypes.string,
+      query: PropTypes.object,
+      search: PropTypes.string,
+    }).isRequired,
+    user: PropTypes.object,
+    history: PropTypes.object,
+    isSideMenuOpened: PropTypes.bool,
+    toggleSideMenu: PropTypes.func,
+  };
+  static defaultProps = {
+    user: {},
+    history: {},
+    isSideMenuOpened: false,
+    toggleSideMenu: () => {},
+  };
+  state = {
+    menuOpened: false,
```

Comment:
Which menu? Sidebar menu? Project selector menu? User menu?
I think it is better to extract this to a corresponding component.

Reply:
fixed